src/pages/trade/model/useSummary.ts

```diff
@@ -24,3 +24,24 @@ export const useSummary = (window: DurationWindow) => {
 
   return query;
 };
+
+// Fetch USDC-denominated prices for each quote asset to enable cross-pair comparisons.
+// Uses 'useQueries' hook for handling multiple dynamic numbers of queries
+export const useMultipleSummaries = (quoteAssets: string[]) => {
+  const queries = useQueries<Array<{ data: SummaryData | NoSummaryData | undefined }>>({
+    queries: quoteAssets.map(asset => ({
+      queryKey: ['multipleSummaries', asset, 'USDC'] as const,
+      queryFn: async () => {
+        return apiFetch<SummaryData | NoSummaryData>('/api/summary', {
+          durationWindow: '1d' as DurationWindow,
+          baseAsset: asset,
+          quoteAsset: 'USDC',
+        });
+      },
+    })),
+  });
+
+  // TODO: add block-based fetching
```

Comment:
we can't simply call `useRefetchOnNewBlock('summary', queries)` since queries is a `UseQueryResult[]`

Reply:
suggestion: something like this would be forward compatible:

```ts
// Use this for your new hook
export const useMultipleRefetchOnNewBlock = (queryKey: string, queries: UseQueryResult[]) => {
  const { data: blockHeight } = useLatestBlockHeight();
  const queryKeyString = JSON.stringify(queryKey);

  useEffect(() => {
    if (!blockHeight) {
      return;
    }

    const lastHeight = lastRefetchedBlockHeights.get(queryKeyString) ?? -1;
    if (blockHeight > lastHeight) {
      lastRefetchedBlockHeights.set(queryKeyString, blockHeight);
      queries.forEach(query => {
        void query.refetch();
      });
    }
  }, [blockHeight, queries, queryKeyString]);
};

// The other ones can still use this
export const useRefetchOnNewBlock = (queryKey: string, query: UseQueryResult) => {
  useMultipleRefetchOnNewBlock(queryKey, [query]);
};
```